### PR TITLE
Mill Module

### LINKED ISSUE #22651: [REQ] Mill Support
Mill is a Scala build tool which can also be used for Java, Kotlin and experimentally for Python, Typescript, Groovy. Its approach is at a sweet spot between Maven (purely declarative) and Gradle (programmable with a lot of plugin magic) since instead of plugins it just relies on tasks and simple libraries. Configuration is done via Scala code or Yaml.

Currently there is only a partial implementation of Openapitools options available in a community project.

Users usually refer to the official tooling for build tool integration, hence providing a official plugin (in Mill just a library module) would make it easy for user to get the support they need.

### PR DESCRIPTION
**Pull request by lostiniceland**
resolves #22651

This PR provides a complete implementation of Openapitool for the [Mill build tool](https://mill-build.org/mill/index.html)

Mill is a Scala build tool which can also be used for Java, Kotlin and experimentally for Python, Typescript, Groovy. Its approach is at a sweet spot between Maven (purely declarative) and Gradle (programmable with a lot of plugin magic) since instead of plugins it just relies on tasks and simple libraries. Configuration is done via Scala code or Yaml.

All config-options from the current Maven & Gradle plugins have been evaluated and added in a similar fashion to keep friction as low as possible. Only for git settings and maven coordinates dedicated case-classes have been added since those configurations belong together.

In contrast to the Gradle plugin, this PR keeps everything under Maven control. So no external Mill executable is run as part of the build. The Mill testkit was used instead for testing via ScalaTest (because ScalaTest has integration to TestNG which is used by Openapitools).
Though the module contains a `mill-test` subproject, this is only meant for manual testing of edge-case configurations and it requires a previous `mvn install` to run properly.

Technical Committee Scala: @clasnake @shijinkui @Bouillie @ramzimaalej 

---
## Summary by cubic
Add first-class Mill support via a new OpenApiModule to generate and validate code from OpenAPI specs, matching Maven/Gradle plugin options. Includes docs, tests, and build wiring for Docker and Maven. Resolves #22651.

- **New Features**
  - New Mill module: modules/openapi-generator-mill-plugin with OpenApiModule and OpenApiConfig.
  - Tasks for generate, validate, and clean; config mirrors Maven/Gradle plugins (git and maven coords as case classes).
  - Runs fully under Maven; uses Mill testkit with ScalaTest; no external Mill process in builds.
  - Example mill-test project for manual edge-case testing (requires mvn install).
  - Added module to root pom and included its pom in Docker go-offline for caching.

- **Docs**
  - Added Mill section to docs/plugins.md with a minimal setup example.
  - New module README with installation, configuration, and usage examples.

<sup>Written for commit 1ca390420a3cf20a7337fc2cb5f41fe748161d41. Summary will update on new commits.</sup>

### CODE REVIEW

File: modules/openapi-generator-mill-plugin/mill-test/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+To test the Mill plugin, this project can be used.
+
+> ![Note]
```
[the remark this answers is not in the document]
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/README.md]: fixed

File: modules/openapi-generator-mill-plugin/mill-test/mill.bat [hunk 2]
```diff
@@ -0,0 +1,299 @@
+@echo off
+
+rem This is a wrapper script, that automatically selects or downloads Mill from Maven Central or GitHub release pages.
+rem
+rem This script determines the Mill version to use by trying these sources
+rem   - env-variable `MILL_VERSION`
+rem   - local file `.mill-version`
+rem   - local file `.config/mill-version`
+rem   - `mill-version` from YAML fronmatter of current buildfile
```
[the remark this answers is not in the document]
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/mill.bat:9]: mill.bat is the official mill script, which is externally maintained. It was just downloaded to execute the demo build and should not be part of the review

File: modules/openapi-generator-mill-plugin/mill-test/api/petstore.yaml [hunk 3]
```diff
@@ -0,0 +1,736 @@
...
+        - store
+      summary: Find purchase order by ID
+      description: >-
+        For valid response try integer IDs with value <= 5 or > 10. Other values
+        will generate exceptions
+      operationId: getOrderById
+      parameters:
+        - name: orderId
+          in: path
+          description: ID of pet that needs to be fetched
```
[the remark this answers is not in the document]
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/api/petstore.yaml:342]: the openapi files in this PR have been copied from external sources and are not meant to be perfect as they are used for testing the module.

File: modules/openapi-generator-mill-plugin/mill-test/mill [hunk 4]
```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env sh
+
+# This is a wrapper script, that automatically selects or downloads Mill from Maven Central or GitHub release pages.
+#
+# This script determines the Mill version to use by trying these sources
+#   - env-variable `MILL_VERSION`
+#   - local file `.mill-version`
+#   - local file `.config/mill-version`
+#   - `mill-version` from YAML fronmatter of current buildfile
```
[the remark this answers is not in the document]
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/mill:9]: the mill shell script  is the official mill script, which is externally maintained. It was just downloaded to execute the demo build and should not be part of the review

File: modules/openapi-generator-mill-plugin/mill-test/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+To test the Mill plugin, this project can be used.
+
+> ![NOTE]
```
[the remark this answers is not in the document]
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/README.md:3]: fixed

**Comment by wing328** [comment]
@lostiniceland thanks for the PR

shall we put it in a separate repo similar to https://github.com/OpenAPITools/sbt-openapi-generator/ ?

**Comment by lostiniceland** [comment]
If you prefer so. For me as a user it is usually just important to see where the integration is coming from and I prefer an official (meaning the source project) over some external integration which is often lacking features.

On the other hand, since the PR is completely based on Maven whereas the SBT plugin uses SBT, I think you are better off with keeping this in the main repo. Its probably better to extract the Gradle plugin from the main repo as it needs workarounds to call Gradle from Maven ;-). 
Split repos tend to increase overhead.

Your repo your call :-)

**Comment by wing328** [comment]
>  Its probably better to extract the Gradle plugin from the main repo as it needs workarounds to call Gradle from Maven ;-).

definitely something we wanted to do it but no one has found the time to do so :(

> For me as a user it is usually just important to see where the integration is coming from and I prefer an official (meaning the source project) over some external integration which is often lacking features.

sounds reasonable

File: modules/openapi-generator-mill-plugin/mill-test/version [hunk 6]
```diff
@@ -0,0 +1 @@
+7.16.0-SNAPSHOT
```
Comment (wing328) [inline modules/openapi-generator-mill-plugin/mill-test/version:1]: quick question: is this file auto generated or need to be updated manually to use the latest version (e.g. 7.19.0-SNAPSHOT) ?

**Comment by wing328** [comment]
when you've time, can you please add a workflow similar to https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/gradle-plugin-tests.yaml to ensure the Mill module is working fine as part of the CI workflow?

File: modules/openapi-generator-mill-plugin/mill-test/version (hunk 6 above)
Comment (lostiniceland) [reply modules/openapi-generator-mill-plugin/mill-test/version:1]: Darn. Forgot to remove this. I tried to somehow get the version from the Maven build over to this test setup without having to set an environment or change the string to get the latest Snapshot build from the local m2.

EDIT: I just found the `release_version_update.sh` script. I will use this to update the version

**Comment by wing328** [comment]
just try to cut a release but failed to publish to maven due to:

```
Error:  

Deployment 7986c10d-d454-4941-9505-efecb3b69c71 failed
pkg:maven/org.openapitools/openapi-generator-mill-plugin@7.19.0:
 - Javadocs must be provided but not found in entries
```
ref: https://github.com/OpenAPITools/openapi-generator/actions/runs/21133323139/job/60776435942

can you please take a look?
